### PR TITLE
added create quay repo + name change to advanced

### PR DESCRIPTION
**Pull request by ElaiShalevRH**
This PR will address the following tasks: 
https://issues.redhat.com/browse/FLPATH-1826
https://issues.redhat.com/browse/FLPATH-1244
https://issues.redhat.com/browse/FLPATH-1834

The changes include adding an additional step in the software-templates; creating a new quay repository in addition to initializing CI/CD. This change includes an additional view-pane for adding configuration to the new quay repository. 

This PR also includes cosmetic name changes: the 'complex-assesment' naming convention was changed to 'advanced'. 

Moreover, some Step titles were shortened and made more concise.

### CODE REVIEW

**Comment by ElaiShalevRH** [comment]
@gabriel-farache 
@masayag

File: scaffolder-templates/advanced-workflow/template.yaml [hunk 1]
```diff
@@ -114,7 +114,7 @@ spec:
                 quayRepoName:
                   title: Quay Repository Name
                   type: string
-                  description: The Quay Repository Name of the published workflow. The repository must exist before deploying the gitops.
+                  description: The Quay Repository Name of the published workflow. If non existent, a new quay repository can be created with this name.
```
Comment (gabriel-farache) [inline scaffolder-templates/advanced-workflow/template.yaml:117]: >If non existent, a new quay repository can be created with this name 


Would rephrase to: 
If non existent, a new quay repository can be created with this name if the parameter `quayCreateNewRepo` is set to `Yes`

or something similar

File: scaffolder-templates/advanced-workflow/template.yaml [hunk 2]
```diff
@@ -162,7 +162,74 @@ spec:
                 - persistencePSQLServiceNamespace
                 - persistencePSQLServicePort
                 - persistencePSQLDatabaseName
+    - title: Create Quay Repository
+      required:
+        - quayCreateNewRepo
+        - CI
+      properties:
+        quayCreateNewRepo:
```
Comment (gabriel-farache) [inline scaffolder-templates/advanced-workflow/template.yaml:170]: I would move all quay related arguments to this "page", like we do for CI/CD (also, rename the title as it will be the quay related page)

Or maybe put `quayCreateNewRepo` in the previous "page" directly so this "page" will be skipped if value set to `No`

**Review by gabriel-farache — CHANGES_REQUESTED** [review]
Nice, good direction; lots of changes, I hope I did not miss anything
All the comments made on the complex workflow applied to the basic one

File: scaffolder-templates/advanced-workflow/template.yaml [hunk 3]
```diff
@@ -294,9 +361,15 @@ spec:
...
         title: Bootstrap the GitOps Resources
+        icon: github
         url: https://github.com/${{ parameters.orgName }}/${{ parameters.repoName }}-gitops/tree/main/bootstrap
       - title: Open the Source Code Repository
+        icon: github
         url: ${{ steps.publishWorkflow.output.remoteUrl }}
       - title: Open the Catalog Info Component
         icon: catalog
         entityRef: ${{ steps.register.output.entityRef }}
+      - if: ${{ parameters.quayCreateNewRepo == 'Yes' }}
```
[the remark this answers is not in the document]
Comment (gabriel-farache) [reply scaffolder-templates/advanced-workflow/template.yaml:372]: let's keep it for consistency at least but put a comment saying the link will appears and that the statement is not working as expected

File: scaffolder-templates/advanced-workflow/template.yaml [hunk 4]
```diff
@@ -162,7 +162,74 @@ spec:
...
+                  type: string
+                  description: The repository desription
+                baseUrl:
+                  title: Quay Base URL
+                  type: string
+                  description: URL of your quay instance, set to "https://quay.io" by default
+                quayNamespace:
+                  title: Quay Namespace
+                  type: string
+                  description: Quay Namespace. Defaults to the namespace of the token.
```
Comment (masayag) [inline scaffolder-templates/advanced-workflow/template.yaml:214]: some descriptions ends with a dot, some aren't. let's go with a common practice for all.

File: scaffolder-templates/advanced-workflow/template.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (masayag) [reply scaffolder-templates/advanced-workflow/template.yaml:372]: also, we can remove all of the `if` from the links section - they are useless.

**Comment by masayag** [comment]
> Nice, good direction; lots of changes, I hope I did not miss anything All the comments made on the complex workflow applied to the basic one

The same goes for my review :-)

File: scaffolder-templates/advanced-workflow/template.yaml [hunk 5]
```diff
@@ -162,7 +162,74 @@ spec:
...
+                  ui:widget: select
+                  enum: ["image", "application"]
+                newRepoDescription:
+                  title: New Quay Repository Description
+                  type: string
+                  description: The repository desription
+                baseUrl:
+                  title: Quay Base URL
+                  type: string
+                  description: URL of your quay instance, set to "https://quay.io" by default
```
Comment (gabriel-farache) [inline scaffolder-templates/advanced-workflow/template.yaml:210]: Do we even need that? Is it possible to have quay repository that are not on quay.io? 
When custom registry will be implemented, maybe it will be possible to have a registry of type `quay`

Comment (ElaiShalevRH) [reply scaffolder-templates/advanced-workflow/template.yaml:210]: I think it's for quay instances that aren't quay.io, not for external reg

Comment (gabriel-farache) [reply scaffolder-templates/advanced-workflow/template.yaml:210]: do you have an example? I never encountered such thing but I am keen to acknowledge their existence

Comment (gabriel-farache) [reply scaffolder-templates/advanced-workflow/template.yaml:210]: >I think it's for quay instances that aren't quay.io, not for external reg
The quay API docs use the term "https://quay-server.example.com/" to refer to the base URL

Ok, I am not familiar with quay, but from what I understand, anyone could deploy a quay registry (hence the `kind` of the custom registry), so let's keep this field

In the PR to use custom registry, we will need to take care of the wording; the custom registry will refer to another kind of registry than `quay`

Comment (ElaiShalevRH) [reply scaffolder-templates/advanced-workflow/template.yaml:210]: I never saw it in person, but [here ](https://www.redhat.com/en/technologies/cloud-computing/quay#:~:text=Ways%20to%20run%20Red%20Hat%20Quay)it looks like it exists

**Comment by masayag** [comment]
@y-first FYI

File: scaffolder-templates/advanced-workflow/template.yaml (hunk 1 above)
Comment (ElaiShalevRH) [reply scaffolder-templates/advanced-workflow/template.yaml]: @gabriel-farache please note the new design:
all quay related information has moved to the Quay 'page'. There, you can select to create a new repository (prompted with all params), use an existing one (prompted for a org and name), or no quay at all (aligns for the case selected no CI method). 
The descriptions for the quay repo name change according to the context.

File: scaffolder-templates/advanced-workflow/template.yaml (hunk 2 above)
Comment (ElaiShalevRH) [reply scaffolder-templates/advanced-workflow/template.yaml]: @gabriel-farache please note the new design:
all quay related information has moved to the Quay 'page'. There, you can select to create a new repository (prompted with all params), use an existing one (prompted for a org and name), or no quay at all (aligns for the case selected no CI method).

File: scaffolder-templates/advanced-workflow/template.yaml [hunk 6]
```diff
@@ -107,62 +108,146 @@ spec:
...
+        - quayConfig
+      properties:
+        quayConfig:
+          title: Create a new Quay repository with the provided name?
+          type: string
+          default: "Use Existing Repository"
+          enum:
+            - "Use Existing Repository"
+            - "Create New Repository"
+            - "No Quay Config Needed"
```
Comment (gabriel-farache) [inline scaffolder-templates/advanced-workflow/template.yaml:167]: If the CI/CD was set to `None` this page should not be displayed to the user, is it possible to add an `if` statement to skip this page?

Otherwise, we should display an empty page with a constant string saying "No CI/CD selected, ignoring quay" or something similar

File: scaffolder-templates/advanced-workflow/template.yaml [hunk 7]
```diff
@@ -107,62 +108,146 @@ spec:
...
+                quayConfig:
+                  const: "Use Existing Repository"
+                quayOrgName:
+                  title: Quay Organization Name
+                  type: string
+                  description: The Quay organization name of the published workflow
+                quayRepoName:
+                  title: Quay Repository Name
+                  type: string
+                  description: The Quay repository Name of the published workflow. The repository must exist before deploying the gitops
```
Comment (gabriel-farache) [inline scaffolder-templates/advanced-workflow/template.yaml:184]: those are duplicated with the later dependencies, just show them all the time

In the later state then, only show the needed extended parameters ot just embed `dependencies` into `dependencies` in the same state

**Review by gabriel-farache — CHANGES_REQUESTED** [review]
Let's rework a bit the quay page and it should be good

**Comment by ElaiShalevRH** [comment]
Hey @masayag @gabriel-farache 
I addressed the UX issues with the Quay page, now all fields are in the Build page, and the flow for the user make more sense. Only drawback is the length of the page, but I believe it is sufficiently divided by the radio widget. 
I resolved all other comments, let me know what you think :)

File: scaffolder-templates/advanced-workflow/template.yaml (hunk 7 above)
Comment (ElaiShalevRH) [reply scaffolder-templates/advanced-workflow/template.yaml]: Duplicating in this way lets me divide the page more nicely to build->quay params. So the Quay name and org fields are after the choice to createquse existing repo, which is more coherent for the user. In addition, this allows me to change the description to match what the user is doing (creating a new repository with the given name, or using an existing).

File: scaffolder-templates/advanced-workflow/template.yaml [hunk 8]
```diff
@@ -107,54 +108,120 @@ spec:
...
+                  description: Port on which the PostgreSQL instance is running
                 persistencePSQLDatabaseName:
                   title: PostgreSQL Database Name
                   type: string
-                  description: Name of the database to use for persistence.
+                  description: Name of the database to use for persistence
                   default: sonataflow
+                quayConfig:
+                  title: Quay Configuration
+                  description: Choose between creating a new Quay repository or using an existing one.
```
Comment (masayag) [inline scaffolder-templates/advanced-workflow/template.yaml:148]: pls remove the dot from the end of the sentence for consistency

File: scaffolder-templates/advanced-workflow/template.yaml [hunk 9]
```diff
@@ -107,54 +108,120 @@ spec:
...
+                          type: string
+                          description: The Quay Organization Name of the published workflow
+                        quayRepoName:
+                          title: Quay Repository Name
+                          type: string
+                          description: A name to create a new Quay Repository of the published workflow
+                        token:
+                          title: Quay AuthToken
+                          type: string
+                          description: Quay rganization bearer token used for authorization
```
Comment (masayag) [inline scaffolder-templates/advanced-workflow/template.yaml:186]: pls fix typo: rganization

File: scaffolder-templates/basic-workflow/template.yaml [hunk 10]
```diff
@@ -137,54 +138,120 @@ spec:
...
+                          type: string
+                          description: The Quay Organization Name of the published workflow
+                        quayRepoName:
+                          title: Quay Repository Name
+                          type: string
+                          description: A name to create a new Quay Repository of the published workflow
+                        token:
+                          title: Quay AuthToken
+                          type: string
+                          description: Quay rganization bearer token used for authorization
```
Comment (masayag) [inline scaffolder-templates/basic-workflow/template.yaml:216]: typo in rganization

**Review by gabriel-farache — APPROVED** [review]

**Review by masayag — APPROVED** [review]
